src/main/java/net/atomique/ksar/ui/NaturalComparator.java

```diff
@@ -39,8 +39,8 @@ public int compare(String a, String b) {
       }
       String u = ma.group(1);
       String v = mb.group(1);
-      boolean isDigit = Character.isDigit(u.charAt(0));
-      if (!isDigit || !Character.isDigit(v.charAt(0))) {
+
+      if (!Character.isDigit(u.charAt(0)) || !Character.isDigit(v.charAt(0))) {
```

Comment:
It looks like there's a way to compare without `parseLong`:

```java
// Both digits => compare the lengths first
// That yields numeric comparison for number-like values
if (Character.isDigit(u.charAt(0)) && Character.isDigit(v.charAt(0))) {
  int res = Integer.compare(u.length(), v.length());
  if (res != 0) {
    // The shorter the length the smaller the number
    return res;
  }
}
return u.compareTo(v);
```

WDYT?

Reply:
looking into this after long time .. your last idea is not fully correct, it was breaking a lot of tests.
The last return should only happen if both numbers are not the same.

```java
      // Both digits => compare the lengths first
      // That yields numeric comparison for number-like values
      if (Character.isDigit(u.charAt(0)) && Character.isDigit(v.charAt(0))) {
        int res = Integer.compare(u.length(), v.length());
        if (res != 0) {
          // The shorter the length the smaller the number
          return res;
        } else {
          if (!u.equals(v)) {
            return u.compareTo(v);
          }
        }
      }
```